src/test/java/teammates/test/cases/testdriver/HtmlHelperTest.java

```diff
@@ -135,4 +135,20 @@ public void testConvertToStandardHtmlPart() throws Exception {
         
         //other cases are tested in testComparison
     }
+    
+    @Test
+    public void testReplacement() throws IOException {
```

Comment:
The risk is minor to non-existent, I believe. I don't agree on "we are not checking whether this result is the CORRECT result". Let's take a look at how the different types of random values are handled:

1. Unpredictable values: Processed only in actual; the one in expected is not touched because it is hard-coded to represent what is CORRECT.
2. Injected values: Processed only in expected; the one in actual is not touched because it is hard-coded (or, more aptly, hard-injected) to represent what is CORRECT. (we really don't need to test `HtmlHelper.injectContextDependentValuesForTest`, right?)
3. Variations (currently only truncated IDs): This might be the problematic one, as it's processed both in actual (suppressing the variation) and expected (injecting the standardized values). I agree that this might warrant a correctness check, but if we look at how it is done currently, omitting won't make it that much weaker (in expected, the values are injected to the same placeholders `${test.account}` as in (2), so if (2) is correct then it's safe to conclude that this will also be).

i.e I don't think adding a HTML comparison after one process makes it _that_ stronger (if any stronger, that is), and one sure thing is it adds the hassle of having to modify one extra HTML file if we modify the algorithm. I might change my mind on (3) if we have a rule that really warrant a correctness check, but at the current state, I see little to no need to.